### ISSUE TITLE
Amplitude for negative frequency

### ISSUE THREAD

**Issue by keiichirokubota** [BlackHolePerturbationToolkit/Teukolsky#49]
I think calculations for negative frequency seem to fail.

For example, a command ``` TeukolskyRadial[-2, 2, 2, 0, 1`30][“In"]["Amplitudes”] ``` returns following results

> <|"Incidence" -> -10.47175832 + 35.36017937 I, 
>  "Transmission" -> 1.0000000000 + 0.*10^-11 I, 
>  "Reflection" -> -0.001473656450 + 0.000629969474 I|>.

Replacing the frequency with a negative value, a command ``` TeukolskyRadial[-2, 2, -2, 0, -1`30][“In"]["Amplitudes”] ``` returns following results

> <|"Incidence" -> -10.47175832 - 35.3601794 I, 
> "Transmission" -> 1.000000000 + 0.*10^-10 I, 
> "Reflection" -> -5.99422984*10^6 - 656978.09 I|>

Using the symmetry of the radial Teukolsky equation under complex conjugate, we find the following relation,
```math 
{}_sR_{lm\omega}={}_sR_{l(-m)(-\omega)}^*. 
```
The above results of reflection amplitude do not seem to satisfy this relation.

**Comment by barrywardell** [comment]
You're right, this seems to be incorrect currently. The evaluation of the solutions at a given radius still works as expected for negative frequencies and the problem does not affect the "up" solutions. As far as I can see, it is only the "in" reflection amplitude that is incorrect. This is computed using Eq. (169) of [Sasaki and Tagoshi](https://doi.org/10.12942/lrr-2003-6). I haven't yet worked out what the problem is, but it must be in the bit in parenthesis involving $K_\nu$ as the rest also appears in the "up" transmission amplitude (Eq. (170)). @znasipak do you have any suggestions?

There is also raises the issue of whether we should even allow negative frequencies. Strictly speaking the "in" solutions are only defined for $\omega > 0$  (and "up" for $k>0$), and for negative frequencies we use the relations to positive frequency solutions. I think we probably should support negative frequencies with this understanding.

**Comment by keiichirokubota** [comment]
Thank you for your reply.

> There is also raises the issue of whether we should even allow negative frequencies. Strictly speaking the "in" solutions are only defined for ω>0 (and "up" for k>0), and for negative frequencies we use the relations to positive frequency solutions. I think we probably should support negative frequencies with this understanding.

I think so. As mentioned in the following sentence of Eq (113) of [Sasaki and Tagoshi](https://doi.org/10.12942/lrr-2003-6),  one can easily obtain a radial function for  negative frequencies from that for positive frequencies. There is no need to calculate for negative frequencies.
